nx/lib/nx.ex

```diff
@@ -4280,6 +4280,90 @@ defmodule Nx do
     )
   end
 
+  @doc """
```

Comment:
@jonatanklosko @josevalim @seanmor5 PR is ready for review. Suggestions for better error messages are more than welcome.

Also, the second example in the docs is kinda useful, but a bit verbose. Should I replace it with something more obvious? Perhaps move that into an unit test of sorts (heh)?